tests/test_clip.py

```diff
@@ -1,127 +1,93 @@
 import subprocess
+from pathlib import Path
 
 import numpy as np
 import pdal
+import pytest
 import utils
 
 
-def test_clip_laz_to_las(main_laz_file: str):
-    """Test clip las function"""
-
-    clipped_las_file = utils.test_data_filepath("clipped.las")
+@pytest.mark.parametrize(
+    "input_type,output",
+    [
+        ("main_laz", utils.test_data_filepath("clipped.las")),
+        ("main_laz", utils.test_data_filepath("clipped.copc.laz")),
+        ("main_copc", utils.test_data_filepath("clipped-copc-input.laz")),
+        ("main_copc", utils.test_data_filepath("clipped-copc-input.copc.laz")),
+    ],
+)
+def test_input_file_output_file(
+    input_type: str,
+    output: Path,
+    main_laz_file: str,
+    main_copc_file: str,
+):
+    """Test clip input is file output is file"""
+
+    if input_type == "main_laz":
+        input = main_laz_file
+    elif input_type == "main_copc":
+        input = main_copc_file
```

Comment:
this parametrization is a bit confusing that there's `input_type` and then based on it we take one of the fixtures... can't we parametrize it so that the parameters are `input` and `output` files? (i don't mind having a bit of duplication of filenames for the sake of simplicity)